awaygoApp/decorators.py

```diff
@@ -0,0 +1,29 @@
+from django.http import HttpResponse
+from django.shortcuts import redirect
+
+def unauthenticated_user(view_func):
+    def wrapper_func(request, *args, **kwargs):
+        if request.user.is_superuser:
+            return redirect('admin/')
+        elif request.user.is_authenticated:
+            return redirect('restaurant')
+        else:
+            return view_func(request, *args, **kwargs)
+
+    return wrapper_func
+
+def allowed_users(allowed_roles={}):
```

Comment:
Because if we modify a default argument, it will persist until the next call, so our "empty" dict will start to contain values on calls other than the first one.

Reply:
True, so you shouldn't use any mutable object as a default argument. You could check if argument is None and assign a default value inside the function instead.